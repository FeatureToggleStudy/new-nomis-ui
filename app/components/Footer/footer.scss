


.FooterContainer{
  position: absolute !important;
  bottom: 0;
  width: 100%;
  background: #DEE0E2;
  padding:1%;
  margin:auto;
}

.FooterLinkContainer{
  padding-top: 10px;
}

.FooterLink{
  float: left;
  padding: 0px 19px;
  border-right: 1px solid #0B0C0C;
  height: 19px;
  cursor: pointer;

  &:last-child {
    border-right: 0px;
  }
  &:first-child {
    padding-left: 14px;
  }
}

.FooterSignature{
  max-width: 1170px;
  clear: left;
  margin-top: 37px;
  padding-left: 14px;
  padding-bottom: 10px;
}