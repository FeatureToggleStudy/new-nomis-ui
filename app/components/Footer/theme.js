import styled from 'styled-components';
import colours from 'theme/colours';
import desktop from 'theme/desktop';
import { mfmedia } from 'utils/style-utils';

export const FooterContainer = styled.div`
  display: none;
  
  ${mfmedia.desktop`display: block;`}
  
  position: absolute;
  bottom: 0;
  width: 100%;
  height: ${desktop.footerHeight}px;
  background: ${colours.footer.bg};
`;

export const FooterLinksContainer = styled.div`
  max-width: ${desktop.fixWidth}px;
  padding-top: 10px;
`;

export const FooterLink = styled.div`
  float: left;
  padding: 0px 19px;
  border-right: 1px solid ${colours.baseFont};
  height: 19px;
  cursor: pointer;

  &:last-child {
    border-right: 0px;
  }
  &:first-child {
    padding-left: 14px;
  }
`;

export const FooterSignature = styled.div`
  max-width: 1170px;
  clear: left;
  margin-top: 37px;
  padding-left: 14px;
  padding-bottom: 10px;
`;
