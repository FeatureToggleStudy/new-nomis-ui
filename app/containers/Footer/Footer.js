import React from 'react'
import {
  Footer,
  Meta,
  MetaItem,
  InlineList,
  LicenseDescription,
  FooterLink,
  LicenseLogo,
  CopyrightLogo,
  HiddenHeader,
} from './Footer.styles'

const FooterGovUK = () => (
  <Footer role="contentinfo">
    <div className="govuk-width-container main-content">
      <Meta>
        <MetaItem grow>
          <HiddenHeader level={2}>Support links</HiddenHeader>
          <InlineList>
            <li>
              <FooterLink href="mailto:feedback@digital.justice.gov.uk" className="govuk-footer__link">
                Contact
              </FooterLink>
            </li>{' '}
            <li>
              <FooterLink href="/auth/terms" id="terms" className="govuk-footer__link">
                Terms and conditions
              </FooterLink>
            </li>
          </InlineList>
          <LicenseLogo
            role="presentation"
            focusable="false"
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 483.2 195.7"
            height="17"
            width="41"
          >
            <path
              fill="currentColor"
              d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            />
          </LicenseLogo>
          <LicenseDescription>
            All content is available under the{' '}
            <FooterLink href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/" rel="license">
              Open Government Licence v3.0
            </FooterLink>
            , except where otherwise stated
          </LicenseDescription>
        </MetaItem>
        <MetaItem>
          <CopyrightLogo href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/">
            Â© Crown copyright
          </CopyrightLogo>
        </MetaItem>
      </Meta>
    </div>
  </Footer>
)

export default FooterGovUK
