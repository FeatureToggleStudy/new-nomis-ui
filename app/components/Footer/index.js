import React from 'react';
import PropTypes from 'prop-types';

import "./footer.scss"

function Footer({ modalData, setModalOpen, setModalData }) {

  const linkClick = (e) => {
    setModalOpen(true);
    setModalData(modalData[e.currentTarget.dataset.name]);
  };

  return (
    <footer className="FooterContainer visible-lg visible-md col-lg-12 col-md-12" data-name={'Footer'}>
      <div className="FooterLinksContainer">
        <div className="FooterLink" data-name={'updates'} onClick={linkClick}>Updates</div>
        <div className="FooterLink" data-name={'help'} onClick={linkClick}>Help</div>
        <div className="FooterLink" data-name={'terms'} onClick={linkClick}>Terms and conditions</div>
      </div>
      <div className="FooterSignature">Powered by Syscon Justice Systems</div>
    </footer>
  );
}

Footer.propTypes = {
  setModalOpen: PropTypes.func.isRequired,
  setModalData: PropTypes.func.isRequired,
  modalData: PropTypes.object.isRequired,
};

Footer.defaultProps = {

};

export default Footer;
