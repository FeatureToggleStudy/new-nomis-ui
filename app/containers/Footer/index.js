import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createSelector, createStructuredSelector } from 'reselect';


import { showTerms } from 'globalReducers/app';

import './footer.scss';

// eslint-disable-next-line jsx-a11y/anchor-has-content
const MailTo = ({ mailTo }) => mailTo ? (<div className="FooterLink">Email&nbsp;&nbsp;<a className="link" href={`mailto:${mailTo}`} />{mailTo}</div>) : null;

const Footer = ({ showTermsAndConditions, mailTo }) =>
      <footer className="FooterContainer">
        <div className="footer-content">
          <div className="FooterLinksContainer">
            <div className="FooterLink" onClick={() => showTermsAndConditions()}>Terms and conditions</div>
            <MailTo mailTo={mailTo} />
          </div>
          <div className="FooterSignature" />
        </div>
      </footer>;

Footer.propTypes = {
  showTermsAndConditions: PropTypes.func.isRequired,
  mailTo: PropTypes.string,
};

Footer.defaultProps = {
};

const selectMailTo = () => createSelector(
  state => state.get('app'),
  appState => appState.get('mailTo')
);

const mapStateToProps = createStructuredSelector({
  mailTo: selectMailTo(),
});

const mapDispatchToProps = (dispatch) => ({
  showTermsAndConditions: () => dispatch(showTerms()),
});

export default connect(mapStateToProps, mapDispatchToProps)(Footer);
