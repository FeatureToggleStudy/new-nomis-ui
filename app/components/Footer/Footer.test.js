import React from 'react'
import renderer from 'react-test-renderer'
import Footer from './Footer'

describe('<Footer />', () => {
  it('should match the default snapshot', () => {
    const wrapper = renderer.create(<Footer />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the with meta snapshot', () => {
    const meta = {
      items: [
        { href: '/help', text: 'Help' },
        { href: '/cookies', text: 'Cookies' },
        { href: '/contact', text: 'Contact' },
        { href: '/terms-conditions', text: 'Terms and conditions' },
        { href: '/welsh', text: 'Rhestr o Wasanaethau Cymraeg' },
      ],
      html: (
        <div>
          Built by the <a href="/gds">Government Digital Service</a>
        </div>
      ),
    }
    const wrapper = renderer.create(<Footer meta={meta} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the with navigation snapshot', () => {
    const navigation = [
      {
        title: 'Services and information',
        columns: 2,
        items: [
          { href: '/benefits', text: 'Benefits' },
          { href: '/births-deaths', text: 'Births, deaths, marriages and care' },
          { href: '/business-self-employed', text: 'Business and self-employed' },
          { href: '/childcare-parenting', text: 'Childcare and parenting' },
          { href: '/citizenship', text: 'Citizenship and living in the UK' },
          { href: '/crime-justice', text: 'Crime, justice and the law' },
          { href: '/disabled-people', text: 'Disabled people' },
          { href: '/driving-transport', text: 'Driving and transport' },
          { href: '/education-learning', text: 'Education and learning' },
          { href: '/employing-people', text: 'Employing people' },
          { href: '/environment-countryside', text: 'Environment and countryside' },
          { href: '/housing-local', text: 'Housing and local services' },
          { href: '/money-tax', text: 'Money and tax' },
          { href: '/passports-abroad', text: 'Passports, travel and living abroad' },
          { href: '/visas-immigration', text: 'Visas and immigration' },
          { href: '/working-pensions', text: 'Working, jobs and pensions' },
        ],
      },
      {
        title: 'Departments and policy',
        items: [
          { href: '/how-gov-works', text: 'How government works' },
          { href: '/departments', text: 'Departments' },
          { href: '/worldwide', text: 'Worldwide' },
          { href: '/policies', text: 'Policies' },
          { href: '/publications', text: 'Publications' },
          { href: '/announcements', text: 'Announcements' },
        ],
      },
    ]
    const wrapper = renderer.create(<Footer navigation={navigation} />).toJSON()

    expect(wrapper).toMatchSnapshot()
  })
})
