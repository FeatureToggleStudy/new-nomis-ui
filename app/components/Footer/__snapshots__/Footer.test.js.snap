// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match the default snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-fhubzv"
  >
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-cy88lo"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            rel="license"
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-juj2a6"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should match the with meta snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-fhubzv"
  >
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <h2
          className="css-a97mk7-StyledHeader-StyledHoc e4u1u2g0"
          size={undefined}
        >
          Support links
        </h2>
        <div
          className="css-z5k0ly"
        >
          <li>
            <a
              className="css-cy88lo"
              href="/help"
            >
              Help
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/cookies"
            >
              Cookies
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/contact"
            >
              Contact
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/terms-conditions"
            >
              Terms and conditions
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/welsh"
            >
              Rhestr o Wasanaethau Cymraeg
            </a>
          </li>
        </div>
        <div
          className="css-1n6hlla"
        >
          <div>
            Built by the 
            <a
              href="/gds"
            >
              Government Digital Service
            </a>
          </div>
        </div>
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-cy88lo"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            rel="license"
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-juj2a6"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;

exports[`<Footer /> should match the with navigation snapshot 1`] = `
<footer
  className="css-1ynevfo"
  role="contentinfo"
>
  <div
    className="css-fhubzv"
  >
    <div
      className="css-thb5ct"
    >
      <div
        className="css-cvoisk"
      >
        <h2
          className="css-1yris73-StyledHeader-StyledHoc e4u1u2g0"
          size="MEDIUM"
        >
          Services and information
        </h2>
        <ul
          className="css-1wreg1n"
        >
          <li>
            <a
              className="css-cy88lo"
              href="/benefits"
            >
              Benefits
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/births-deaths"
            >
              Births, deaths, marriages and care
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/business-self-employed"
            >
              Business and self-employed
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/childcare-parenting"
            >
              Childcare and parenting
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/citizenship"
            >
              Citizenship and living in the UK
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/crime-justice"
            >
              Crime, justice and the law
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/disabled-people"
            >
              Disabled people
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/driving-transport"
            >
              Driving and transport
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/education-learning"
            >
              Education and learning
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/employing-people"
            >
              Employing people
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/environment-countryside"
            >
              Environment and countryside
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/housing-local"
            >
              Housing and local services
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/money-tax"
            >
              Money and tax
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/passports-abroad"
            >
              Passports, travel and living abroad
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/visas-immigration"
            >
              Visas and immigration
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/working-pensions"
            >
              Working, jobs and pensions
            </a>
          </li>
        </ul>
      </div>
      <div
        className="css-cvoisk"
      >
        <h2
          className="css-1yris73-StyledHeader-StyledHoc e4u1u2g0"
          size="MEDIUM"
        >
          Departments and policy
        </h2>
        <ul
          className="css-ul2glb"
        >
          <li>
            <a
              className="css-cy88lo"
              href="/how-gov-works"
            >
              How government works
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/departments"
            >
              Departments
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/worldwide"
            >
              Worldwide
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/policies"
            >
              Policies
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/publications"
            >
              Publications
            </a>
          </li>
          <li>
            <a
              className="css-cy88lo"
              href="/announcements"
            >
              Announcements
            </a>
          </li>
        </ul>
      </div>
    </div>
    <hr
      className="css-uf3sj0"
    />
    <div
      className="css-dzhhuw"
    >
      <div
        className="css-1g6dja6"
      >
        <svg
          className="css-11k3glg"
          focusable="false"
          height="17"
          role="presentation"
          viewBox="0 0 483.2 195.7"
          width="41"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145"
            fill="currentColor"
          />
        </svg>
        <span
          className="css-7kp13n"
        >
          All content is available under the
           
          <a
            className="css-cy88lo"
            href="https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/"
            rel="license"
          >
            Open Government Licence v3.0
          </a>
          , except where otherwise stated
        </span>
      </div>
      <div
        className="css-3cr27l"
      >
        <a
          className="css-juj2a6"
          href="https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/"
        >
          © Crown copyright
        </a>
      </div>
    </div>
  </div>
</footer>
`;
